src/iterative_sorting/iterative_sorting.py

```diff
@@ -1,25 +1,33 @@
 # TO-DO: Complete the selection_sort() function below 
+
 def selection_sort( arr ):
     # loop through n-1 elements
     for i in range(0, len(arr) - 1):
         cur_index = i
         smallest_index = cur_index
-        # TO-DO: find next smallest element
-        # (hint, can do in 3 loc) 
-             
-
-
-
-        # TO-DO: swap
-
-
-
+        # TO-DO: find next smallest element (hint, can do in 3 loc - lines of code) 
+        # First, I need to iterate through the elements that remain in the arr
+        for x in range(smallest_index + 1, len(arr)):
+            # Second, I need to check if the element is smaller than the currently smallest
+            if arr[x] < arr[smallest_index]:
+                # Third, if this statement is True, I will need to set the new index to be the smallest
+                smallest_index = x
+        # TO-DO: swap the arr
+        arr[cur_index], arr[smallest_index] = arr[smallest_index], arr[cur_index]
 
     return arr
 
 
 # TO-DO:  implement the Bubble Sort function below
+
```

Comment:
Think about how you can optimize this so that it doesn't keep looping through a sorted array. How would you modify your outer loop? How would you check if the array is sorted?